File: basketball_backend/models/Game.js
// models/Game.js
const mongoose = require('mongoose');

const PlayerStatsSchema = new mongoose.Schema({
  player: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Player',
    required: true,
  },
  minutesPlayed: { type: Number, default: 0 },
  pointsMade: { type: Number, default: 0 },
  shotsAttempted: { type: Number, default: 0 },
  shotsMade: { type: Number, default: 0 },
  turnovers: { type: Number, default: 0 },
  steals: { type: Number, default: 0 },
  defensiveRebounds: { type: Number, default: 0 },
  offensiveRebounds: { type: Number, default: 0 },
  foulsCommitted: { type: Number, default: 0 },
  foulsReceived: { type: Number, default: 0 },
  assists: { type: Number, default: 0 },
}, { _id: false });

const GameSchema = new mongoose.Schema({
  date: {
    type: Date,
    required: true,
  },
  homeTeam: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Team',
    required: true,
  },
  awayTeam: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Team',
    required: true,
  },
  homeScore: {
    type: Number,
    required: true,
  },
  awayScore: {
    type: Number,
    required: true,
  },
  playerStats: [PlayerStatsSchema], // Nuovo campo
  createdBy: { // Nuovo campo
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
}, { timestamps: true });

const Game = mongoose.model('Game', GameSchema);

module.exports = Game;

File: basketball_stats_app/lib/models/game.dart
// lib/models/game.dart
import 'package:json_annotation/json_annotation.dart';
import 'team.dart';
import 'player_stats.dart'; // Nuovo modello

part 'game.g.dart';

@JsonSerializable(explicitToJson: true)
class Game {
  @JsonKey(name: '_id')
  final String? id;
  final DateTime date;
  final Team homeTeam;
  final Team awayTeam;
  final int homeScore;
  final int awayScore;
  final List<PlayerStats>? playerStats; // Nuovo campo
  final String? createdBy; // Nuovo campo
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Game({
    this.id,
    required this.date,
    required this.homeTeam,
    required this.awayTeam,
    required this.homeScore,
    required this.awayScore,
    this.playerStats,
    this.createdBy,
    this.createdAt,
    this.updatedAt,
  });

  factory Game.fromJson(Map<String, dynamic> json) => _$GameFromJson(json);
  Map<String, dynamic> toJson() => _$GameToJson(this);
}
